src/framework/audio/internal/platform/win/audiodeviceslistener.cpp

```diff
@@ -118,11 +118,11 @@ HRESULT AudioDevicesListener::OnDefaultDeviceChanged(EDataFlow flow, ERole role,
         return S_OK;
     }
 
-    if (m_previousIdString == new_default_device_id) {
+    if (m_previousDefaultDeviceId == new_default_device_id) {
```

Comment:
Now that `m_previousDefaultDeviceId` and `new_default_device_id` are both plain `LPCWSTR` (aka `wchar_t const *`), this simply compares the pointers, and does not compare the strings. Is that desired, or should maybe `lstrcmpW` be used instead?

Reply:
Microsoft docs say the pointer is only valid for the duration of the call -- so you will need to copy the string into something and do a full string compare.

Reply:
ok, looks like initializing hstring from NULL doesn't work at all. I just added a check as recommended in the issue